### PR TITLE
Added documentation for modify time in a bunch of places

### PR DESCRIPTION
**Pull request by preetapan**
I may be missing more places in the documentation beyond these, so pls take a look

### CODE REVIEW

File: website/source/docs/commands/job/status.html.md.erb [hunk 1]
```diff
@@ -22,7 +22,9 @@ the specific job is queried and displayed. Otherwise, a list of matching jobs an
 information will be displayed.
 
 If the ID is omitted, the command lists out all of the existing jobs and a few of
-the most useful status fields for each.
+the most useful status fields for each. As of Nomad 0.7.1, alloc status also shows allocation
+modification time in addition to create time. When the -verbose flag is not set, allocation
+creation and modify times are shown in a shortened relative time format like `5m ago`.
```
Comment (schmichael) [inline website/source/docs/commands/job/status.html.md.erb]: :tada:

File: website/source/docs/commands/job/status.html.md.erb [hunk 2]
```diff
@@ -22,7 +22,9 @@ the specific job is queried and displayed. Otherwise, a list of matching jobs an
 information will be displayed.
 
 If the ID is omitted, the command lists out all of the existing jobs and a few of
-the most useful status fields for each.
+the most useful status fields for each. As of Nomad 0.7.1, alloc status also shows allocation
+modification time in addition to create time. When the -verbose flag is not set, allocation
```
Comment (schmichael) [inline website/source/docs/commands/job/status.html.md.erb]: Put ticks around -verbose:

```
`-verbose`
```

**Comment by dadgar** [comment]
Also here: https://www.nomadproject.io/intro/getting-started/jobs.html

**Comment by preetapan** [comment]
Added to jobs/nodes/deployments pages that listed allocations.

**Pull request merged**